torch/csrc/distributed/c10d/comm.h

```diff
@@ -45,7 +45,7 @@ class TORCH_API CommHookInterface {
   // Runs the registered communication hook to communicate gradients
   // asynchronously, Returns a future that holds the communication results.
   virtual c10::intrusive_ptr<torch::jit::Future> runHook(
-      const GradBucket& bucket) = 0;
+      GradBucket& bucket) = 0;
```

Comment:
What is the reason to remove const from here?

Reply:
It's because of the subsequent change, which updates the bucket in place.